ceph/ceph_admin/helper.py

```diff
@@ -831,6 +860,49 @@ def validate_log_file_after_enable(cls):
     return True
 
 
+def validate_log_rotate(cls):
+    """
+    Verify logrotation of cephadm.log file
+    Args:
+        cls: cephadm instance object
+
+    Returns:
+        boolean
+    """
+    # step 1: check for the files in current log directory
+    if not file_or_path_exists(cls.installer, "/var/log/ceph/cephadm.log"):
+        LOG.error("Log rotation failed: Logs not present in /var/log/ceph folder")
+        return False
+
+    # step 2: apply logrotate on cephadm.log
+    out = cls.installer.exec_command(
+        cmd="logrotate --force /etc/logrotate.d/cephadm", sudo=True
+    )
+    # if logrotate doesn't result in logs being zipped return False
+    if not file_or_path_updated(
+        cls.installer, "/var/log/ceph/cephadm.log.1.gz", 10
```

Comment:
are you sure the zipped file after log rotate it will be this `cephadm.log.1.gz` name?

Reply:
Yes sunil, I have tested it and it will be. Everytime the logrotate command renames the old files and creates new file with name cephadm.log.1.gz itself. The old ones will be moved to cephadm.log.2.gz and etc.,

Reply:
What if this file is already present? In this case we are looking into incorrect file.

Reply:
As discussed the new file will always be cephadm.log.1.gz